### PR TITLE
Add post-release workflow to publish snap to snap store

### PR DESCRIPTION
**Pull request by defunctzombie**
**User-Facing Changes**
None

**Description**
Publish the snap file to snap store post release.

### CODE REVIEW

**Comment by defunctzombie** [comment]
@amacneil How did you test these post-release workflows?

File: .github/workflows/post-release.yml [hunk 1]
```diff
@@ -181,3 +181,32 @@ jobs:
...
+    name: Publish snap
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout ${{ github.event.release.target_commitish || github.event.inputs.target_commitish }}
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.release.target_commitish || github.event.inputs.target_commitish }}
+          token: ${{ secrets.FOXGLOVEBOT_GITHUB_TOKEN }}
+          lfs: true
```
Comment (amacneil) [inline .github/workflows/post-release.yml:195]:
```suggestion
      - name: Checkout ${{ github.event.release.tag_name || github.event.inputs.tag_name }}
        uses: actions/checkout@v3
        with:
          ref: ${{ github.event.release.tag_name || github.event.inputs.tag_name }}
          token: ${{ secrets.FOXGLOVEBOT_GITHUB_TOKEN }}
          lfs: true
```

File: .github/workflows/post-release.yml [hunk 2]
```diff
@@ -181,3 +181,32 @@ jobs:
...
+
+      - uses: actions/setup-node@v3.3.0
+        with:
+          node-version: 14.x
+          cache: yarn
+
+      - run: yarn install --immutable
+
+      - name: Download release binaries
+        run: yarn ts-node downloadReleases.ts
```
Comment (amacneil) [inline .github/workflows/post-release.yml:205]:
```suggestion
      - name: Download release binaries
        working-directory: ci/apt
        run: yarn ts-node downloadReleases.ts
```

File: .github/workflows/post-release.yml [hunk 3]
```diff
@@ -181,3 +181,32 @@ jobs:
...
+
+      - uses: snapcore/action-publish@v1
+        env:
+          SNAPCRAFT_STORE_CREDENTIALS: ${{ secrets.SNAPCRAFT_STORE_LOGIN }}
+        with:
+          snap: fixme-path-to-file.snap
```
Comment (amacneil) [inline .github/workflows/post-release.yml:211]: ?

File: ci/apt/downloadReleases.ts [hunk 4]
```diff
@@ -31,7 +31,7 @@ async function main(): Promise<void> {
     .slice(0, NUM_VERSIONS_TO_PUBLISH)) {
     for (const asset of release.assets) {
       // Download debs for this release
-      if (asset.name.match(/\.deb$/)) {
+      if (asset.name.match(/\.deb$/) || asset.name.match(/\.snap$/)) {
```
Comment (amacneil) [inline ci/apt/downloadReleases.ts:34]:
```suggestion
      if (asset.name.match(/\.(deb|snap)$/)) {
```

**Comment by amacneil** [comment]
> @amacneil How did you test these post-release workflows?

ship it and cross your fingers.

or you can manually trigger it for a previous release, although most of the actions will only work once (e.g. npm can't overwrite existing releases)

File: .github/workflows/post-release.yml (hunk 2 above)
Comment (defunctzombie) [reply .github/workflows/post-release.yml:205]: I don't want `ci/apt` as the working directory

Comment (amacneil) [reply .github/workflows/post-release.yml:205]: well that's too bad, because that is where the `downloadReleases.ts` script lives

Comment (defunctzombie) [reply .github/workflows/post-release.yml:205]: oh I see - so I can change this to `yarn ts-node cr/apt/downloadReleasese.ts`

Comment (defunctzombie) [reply .github/workflows/post-release.yml:205]: I need to figure out where the actual snap files end up and what they are called so I can use the snap upload action.

**Comment by defunctzombie** [comment]
@amacneil This is ready for review

**Pull request merged**